### PR TITLE
Implement youtube widget real functionality

### LINKED ISSUE #245: Implement basic youtube feature/side widget

### LINKED ISSUE #260: Implement youtube widget real functionality
Allow the user to create a spotify popup and play music

### PR DESCRIPTION
**Pull request by bche536**
Added in the youtube sidewidget drawer button to create a youtube player widget (WidgetYoutubePlayer.jsx)

Created youtube player component (JournalYoutubePlayer.jsx) which contains the actual youtube iframe (created by Aniket)

Added in widget data and type in Page.js

Added in widget type to widget.js in the backend

Added resizable styling and reordered components

Co-authored-by: Aniket

### CODE REVIEW

**Review by aniket-rai — CHANGES_REQUESTED** [review]
Overall looks good. Glad to see my code worked as expected as well. Some changes regarding styling need to be made. 

These are:
- Alignment of textbook to youtube iframe
- Display placement of initial widget
- Input text box display changes and formatting

Let's book a time and work on this together to get it sorted before merging into production.

**Review by aniket-rai — CHANGES_REQUESTED** [review]
Sweet! That works as expected -- thanks for taking the time to peer program with me. Final change would be to change the ID parsing to have URLs instead of just IDs. Will pass this code to you for a final commit.

**Comment by bche536** [comment]
> Sweet! That works as expected -- thanks for taking the time to peer program with me. Final change would be to change the ID parsing to have URLs instead of just IDs. Will pass this code to you for a final commit.

ok

**Review by aniket-rai — APPROVED** [review]
LGTM! Just for documentation - over discord, we discussed removing the submit button and have a default video ready for the user. Extra CSS was done to ensure correct resizing and responsiveness. These were completed in addition to the integration of the function I wrote to parse IDs from URLs.

File: telos-frontend/src/components/youtube-player/JournalYoutubePlayer.jsx [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+// //           Submit
+// //         </button>
+// //       )}
+// //       <input
+// //         className={styles.inputField}
+// //         onChange={(e) => setVidID(YouTubeGetID(e.target.value))}
+// //       />
+// //     </div>
+// //   );
+// // }
```
Comment (PJhaveri02) [inline telos-frontend/src/components/youtube-player/JournalYoutubePlayer.jsx]: I think you should probably remove these comments

**Review by PJhaveri02 — CHANGES_REQUESTED** [review]
Overall, this PR is pretty good. Just remove the comments in JournalYoutubePlayer.jsx and then it should be good to merge.

File: telos-frontend/src/components/youtube-player/JournalYoutubePlayer.jsx (hunk 1 above)
Comment (bche536) [reply telos-frontend/src/components/youtube-player/JournalYoutubePlayer.jsx]: haha oops! will change it now

**Review by bche536 — COMMENTED** [review]
Removed commented code

**Comment by bche536** [comment]
> Overall, this PR is pretty good. Just remove the comments in JournalYoutubePlayer.jsx and then it should be good to merge.

done that now

**Review by PJhaveri02 — APPROVED** [review]
Nice Work! Required changes have been made.

**Pull request merged**